src/controllers/erc721/metadata/multiple/post.controller.ts

```diff
@@ -0,0 +1,126 @@
+    const zip = createArchiver().jsZip;
+    // #swagger.tags = ['ERC721']
+    const erc721 = await ERC721Service.findById(req.params.id);
+    if (!erc721) throw new NotFoundError('Could not find this NFT in the database');
+
+    // UNZIP THE FILE
+    const metadatas: ERC721MetadataDocument[] = [];
+
+    // LOAD ZIP IN MEMORY
+    const contents = await zip.loadAsync(req.file.buffer);
+
+    // ITERATE THE CONTENT BY OBJECT KEYS
+    const objectKeys = Object.keys(contents.files);
+    const promises = [];
+
+    for (let i = 0; i < objectKeys.length; i++) {
+        const file = objectKeys[i];
+        const [originalFileName, extension] = file.split('.');
+
+        if (!extension) {
+            continue;
+        }
+        // FILE VALIDATION
+        if (!isValidExtension(extension)) {
+            logger.info(`INVALID EXTENSION, FILE SKIPPED: ${file}`);
+            continue;
+        }
+        // CREATE THE FILE BUFFER
+        const buffer = await zip.file(file).async('nodebuffer');
+
+        if (!(await isValidFileType(buffer))) {
+            logger.info(`INVALID FILE TYPE, FILE SKIPPED: ${file}`);
+            continue;
+        }
+
+        const promise = new Promise(async (resolve, reject) => {
+            try {
+                // FORMAT FILENAME
+                const filename =
+                    originalFileName.toLowerCase().split(' ').join('-').split('.') +
+                    '-' +
+                    short.generate() +
+                    `.${extension}`;
+
+                // PREPARE PARAMS FOR UPLOAD TO S3 BUCKET
+                const uploadParams = {
+                    Key: filename,
+                    Bucket: AWS_S3_PUBLIC_BUCKET_NAME,
+                    ACL: 'public-read',
+                    Body: buffer,
+                };
+
+                // UPLOAD THE FILE TO S3
+                await s3Client.send(new PutObjectCommand(uploadParams));
+
+                // COLLECT THE URL
+                const url = ImageService.getPublicUrl(filename);
+
+                // CREATE THE METADATA
+                const metadata = await ERC721Service.createMetadata(erc721, req.body.title, req.body.description, [
+                    { key: req.body.propName, value: url },
+                ]);
+
+                metadatas.push(metadata);
+                resolve(metadata);
+            } catch (err) {
+                logger.error(err);
+                reject(err);
+            }
+        });
+
+        promises.push(promise);
+    }
+    await Promise.all(promises);
+    res.status(201).json({ metadatas });
+};
+
+function isValidExtension(extension: string) {
+    const allowedExtensions = ['jpg', 'jpeg', 'gif', 'png'];
+    return allowedExtensions.includes(extension);
+}
+
+async function isValidFileType(buffer: Buffer) {
+    const { mime } = await fromBuffer(buffer);
+    const allowedMimeTypes = ['image/jpeg', 'image/png', 'image/gif'];
+
+    if (!allowedMimeTypes.includes(mime)) {
```

Comment:
@valeriagrazzini  This could just be 2 lines of code right?
```
const { mime } = await fromBuffer(buffer);
return ['image/jpeg', 'image/png', 'image/gif'].includes(mime)
```